src/components/structures/RoomView.js

```diff
@@ -95,15 +103,14 @@ module.exports = React.createClass({
 
     getInitialState: function() {
         var room;
+        var room_id;
         if (this.props.roomAddress[0] == '!') {
-            room = MatrixClientPeg.get().getRoom(this.props.roomAddress);
-        } else {
-            room = MatrixTools.getRoomForAlias(
-                MatrixClientPeg.get().getRooms(), this.props.roomAddress
-            );
+            room_id = this.props.roomAddress;
+            room = MatrixClientPeg.get().getRoom(room_id);
         }
         return {
             room: room,
+            roomId: room_id,
```

Comment:
wouldn't mind seeing some comments about when these are set and how they interact.